### PR TITLE
Set stereo=1 on subscriber PC also

### PR DESCRIPTION
**Pull request by boks1971**
Have asked @bekriebel and @mmckegg to try this branch. Let's see if this helps. Setting `stereo=1` in down stream also.

### CODE REVIEW

**Review by davidzhao — APPROVED** [review]

**Comment by boks1971** [comment]
Have confirmation from @bekriebel that it works without `opus/red`, i. e. using Firefox on receive side makes it work without the change introduced in this PR. So, will probably end up closing this after learning a bit more conditions under which it works/does not work.

File: pkg/rtc/transport.go [hunk 1]
```diff
@@ -1007,17 +1007,50 @@ func (t *PCTransport) configureReceiverDTXAndStereo(enableDTX bool) {
...
+		if sender == nil {
+			continue
 		}
 
-		err := transceiver.SetCodecPreferences(append(modifiedReceiverCodecs, senderCodecs...))
+		var modifiedSenderCodecs []webrtc.RTPCodecParameters
+
+		senderCodecs := sender.GetParameters().Codecs
+		for _, senderCodec := range senderCodecs {
+			if senderCodec.MimeType == webrtc.MimeTypeOpus {
```
Comment (davidzhao) [inline pkg/rtc/transport.go:1036]: is it because of this check? I think MimeType is `audio/red` for Opus RED

Comment (bekriebel) [reply pkg/rtc/transport.go:1036]: I just tried changing it, and the line on recieverCodec to be:
`if receiverCodec.MimeType == webrtc.MimeTypeOpus || receiverCodec.MimeType == "audio/red" {`
`if senderCodec.MimeType == webrtc.MimeTypeOpus || senderCodec.MimeType == "audio/red" {`

It didn't seem to make a difference. There's more discussion on #970 as well.

**Comment by mmckegg** [comment]
My branch that adds this client side does indeed work, and that’s along side red in chrome, so we must be close to it now. 

Not in front of my computer right now, but I can try poking around later.

**Comment by bekriebel** [comment]
> My branch that adds this client side does indeed work, and that’s along side red in chrome, so we must be close to it now.
> 
> Not in front of my computer right now, but I can try poking around later.

I can confirm that using https://github.com/livekit/client-sdk-js/pull/450 and having the `RoomOption` `forceStereoAudioSupport: true` works for Chrome <-> Chrome stereo support.

**Comment by cnderrauber** [comment]
stereo is a receiver side attribute, so don't need add it at subscriber pc in theory. I don't have a stereo mic now, so the change of stereo is rely on my experience. @bekriebel 's result is interesting, will dig into it

**Comment by bekriebel** [comment]
> I don't have a stereo mic now, so the change of stereo is rely on my experience. @bekriebel 's result is interesting, will dig into it 

An easy test I was doing for a while is to test with a screen share. Open a tab with youtube, load up a stereo test video, mute the tab, screen share the tab with audio with the correct capture and publisher options. A muted tab will still share audio through webrtc and can come through as stereo.

**Comment by mmckegg** [comment]
> An easy test I was doing for a while is to test with a screen share. 

Yep this is also what I’ve been doing.

**Comment by bekriebel** [comment]
This isn't needed, but a chance will be needed to the client SDK. I'll put more information in #970 momentarily.

**Comment by boks1971** [comment]
This is not the right fix for this. Closing.

**Pull request closed without merging**